src/features/currentTodo.ts

```diff
@@ -1,10 +1,19 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
 const initialState = null as Todo | null;
 
 export const currentTodoSlice = createSlice({
   name: 'currentTodo',
   initialState,
-  reducers: {},
+  reducers: {
+    add: (state, action: PayloadAction<Todo>) => {
+      state = action.payload;
```

Comment:
In the `add` reducer, the state should be returned instead of directly assigning it. Change `state = action.payload;` to `return action.payload;` to ensure the state is updated correctly.